mlrun/projects/project.py

```diff
@@ -3782,6 +3782,24 @@ def set_model_monitoring_credentials(
             self.register_datastore_profile(stream_profile)
             stream_profile_name = stream_profile.name
 
+            if mlrun.mlconf.artifact_path.startswith("v3io://"):
+                profile = mlrun.datastore.datastore_profile.DatastoreProfileV3io(
+                    name=mm_constants.DefaultProfileName.PARQUET,
+                    v3io_access_key=mlrun.get_secret_or_env("V3IO_ACCESS_KEY"),
+                )
+                self.register_datastore_profile(profile)
+            elif mlrun.mlconf.artifact_path.startswith("s3://"):
+                profile = mlrun.datastore.datastore_profile.DatastoreProfileS3(
+                    name=mm_constants.DefaultProfileName.PARQUET,
+                    access_key_id=mlrun.get_secret_or_env("AWS_ACCESS_KEY_ID"),
+                    secret_key=mlrun.get_secret_or_env("AWS_SECRET_ACCESS_KEY"),
+                    endpoint_url=mlrun.get_secret_or_env("S3_ENDPOINT_URL"),
+                    force_non_anonymous=mlrun.get_secret_or_env("S3_NON_ANONYMOUS"),
+                    profile_name=mlrun.get_secret_or_env("AWS_PROFILE"),
+                    assume_role_arn=mlrun.get_secret_or_env("MLRUN_AWS_ROLE_ARN"),
+                )
+                self.register_datastore_profile(profile)
```

Comment:
I don't really like this solution for two reasons:

1. You rely on the existence of the environment variables in the client's SDK environment. However, many times they are just not there.
2. `set_model_monitoring_credentials` should not create or register datastore profiles under the hood. The current creation and registration is a transitionary state, and will be removed soon.